warp/backend.go

```diff
@@ -56,11 +59,13 @@ type backend struct {
 	messageSignatureCache *cache.LRU[ids.ID, [bls.SignatureLen]byte]
 	blockSignatureCache   *cache.LRU[ids.ID, [bls.SignatureLen]byte]
 	messageCache          *cache.LRU[ids.ID, *avalancheWarp.UnsignedMessage]
+	offChainMessageMap    map[ids.ID]*avalancheWarp.UnsignedMessage
 }
 
 // NewBackend creates a new Backend, and initializes the signature cache and message tracking database.
-func NewBackend(networkID uint32, sourceChainID ids.ID, warpSigner avalancheWarp.Signer, blockClient BlockClient, db database.Database, cacheSize int) Backend {
-	return &backend{
+// It returns an error if [offChainMessages] is invalid.
+func NewBackend(networkID uint32, sourceChainID ids.ID, warpSigner avalancheWarp.Signer, blockClient BlockClient, db database.Database, cacheSize int, offChainMessages []hexutil.Bytes) (Backend, error) {
```

Comment:
```suggestion
func NewBackend(networkID uint32, sourceChainID ids.ID, warpSigner avalancheWarp.Signer, blockClient BlockClient, db database.Database, cacheSize int, offChainMessages [][]byte) (Backend, error) {
```

Let's convert from `[]hexutil.Bytes` to `[][]byte` in `vm.Initialize(...)` instead of passing in the type alias `hexutil.Bytes` here

Reply:
done